packages/aws-rfdk/lib/core/scripts/powershell/configureCloudWatchAgent.ps1

```diff
@@ -16,62 +16,90 @@ param (
 
 $ErrorActionPreference = "Stop"
 
-$file = "$env:temp\amazon-cloudwatch-agent.msi"
 
-# We hardcode the version number here since the verification below is tightly coupled with this specific version
+Write-Output "Beginning CloudWatch agent verification and installation."
+
+$cwa_installer = "$env:temp\amazon-cloudwatch-agent.msi"
 try {
-  Read-S3Object -BucketName amazoncloudwatch-agent -Key windows/amd64/1.242486.0/amazon-cloudwatch-agent.msi -File $file -Region us-east-1
+  Read-S3Object -BucketName amazoncloudwatch-agent -Key windows/amd64/latest/amazon-cloudwatch-agent.msi -File $cwa_installer -Region us-east-1
 } catch {
-  # Fallback to the latest version (this is the case when the above version is currently "latest")
-  Write-Output "Initial attempt to download Amazon CloudWatch agent failed. Falling back to the latest version."
-  try {
-    Read-S3Object -BucketName amazoncloudwatch-agent -Key windows/amd64/latest/amazon-cloudwatch-agent.msi -File $file -Region us-east-1
-  } catch {
-    Write-Output "Failed to download CloudWatchAgent installer."
-    Exit 1
-  }
+  Write-Output "Failed to download CloudWatch agent installer."
+  Exit 1
+}
+
+$cwa_installer_sig = "$env:temp\amazon-cloudwatch-agent.msi.sig"
+try {
+  Read-S3Object -BucketName amazoncloudwatch-agent -Key windows/amd64/latest/amazon-cloudwatch-agent.msi.sig -File $cwa_installer_sig -Region us-east-1
+} catch {
+  Write-Output "Failed to download CloudWatch agent installer signature file."
+  Exit 1
 }
 
-# The below verification is tied to the version of the msi installer
-# If the installer version changes, the below verification could break and should be updated accordingly.
-# See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/verify-CloudWatch-Agent-Package-Signature.html
 if (-Not $s) {
-  $sig = Get-AuthenticodeSignature $file
-  $status = $sig | Select-Object -ExpandProperty 'Status'
+  # Download GPG
+  $gpg_installer = "$env:temp\gnupg-w32-2.2.27_20210111.exe"
+  wget https://gnupg.org/ftp/gcrypt/binary/gnupg-w32-2.2.27_20210111.exe -OutFile $gpg_installer
+
+  # Verify GPG
+  $gpg_sig = Get-AuthenticodeSignature $gpg_installer
+  $status = $gpg_sig | Select-Object -ExpandProperty 'Status'
   if ( $status -ne 'Valid' ) {
-    Write-Output "CloudWatchAgent installer does not have a valid signature."
+    Write-Output "GPG installer does not have a valid signature."
     Exit 1
   }
```

Comment:
Not sure if we need to check this since we don't check any other parts of the Authenticode signature. We could check the certificates like we did in the [old version](https://github.com/aws/aws-rfdk/blob/mainline/packages/aws-rfdk/lib/core/scripts/powershell/configureCloudWatchAgent.ps1#L50), but I think just checking the sha1 checksum should be enough.